receiver/otlpreceiver/otlp_test.go

```diff
@@ -384,6 +385,87 @@ func TestHandleInvalidRequests(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestBadHTTPJSONRequest(t *testing.T) {
+	endpoint := testutil.GetAvailableLocalAddress(t)
+	cfg := &Config{
+		Protocols: Protocols{
+			HTTP: &HTTPConfig{
+				HTTPServerSettings: &confighttp.HTTPServerSettings{
+					Endpoint: endpoint,
+				},
+				TracesURLPath:  defaultTracesURLPath,
+				MetricsURLPath: defaultMetricsURLPath,
+				LogsURLPath:    defaultLogsURLPath,
+			},
+		},
+	}
+
+	// Traces
+	tr, err := NewFactory().CreateTracesReceiver(
+		context.Background(),
+		receivertest.NewNopCreateSettings(),
+		cfg,
+		consumertest.NewNop())
+	require.NoError(t, err)
+	assert.NotNil(t, tr)
+	require.NoError(t, tr.Start(context.Background(), componenttest.NewNopHost()))
+	t.Cleanup(func() {
+		require.NoError(t, tr.Shutdown(context.Background()))
+	})
+
+	traceJSON := `{
```

Comment:
What is bad about this request?

Reply:
the span is invalid as its fields are encoded with camel case instead of snake case, such as `traceId` instead of `trace_id`.

Reply:
https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/specification.md#json-protobuf-encoding

Reply:
This is not an invalid message per the definition of the protocol.

Reply:
ok sorry. https://github.com/open-telemetry/opentelemetry-collector/issues/6320#issuecomment-1279738087 I was looking in the wrong place.

Reply:
That comment says that we cannot do anything unfortunately for that.